gtsfm/data_association/data_assoc.py

```diff
@@ -270,6 +272,42 @@ def triangulate_batch(
 
         return sfm_tracks, avg_track_reproj_errors, triangulation_exit_codes
 
+    def run_triangulation_and_evaluate(
+        self,
+        num_images: int,
+        cameras: Dict[int, gtsfm_types.CAMERA_TYPE],
+        tracks_2d: List[SfmTrack2d],
+        cameras_gt: List[Optional[gtsfm_types.CAMERA_TYPE]],
+        relative_pose_priors: Dict[Tuple[int, int], PosePrior],
+        images: Optional[List[Delayed]] = None,
+    ) -> Tuple[GtsfmData, GtsfmMetricsGroup]:
+        """Runs triangulation, evaluates results, and forms metrics group."""
+        start_time = time.time()
+
+        # Triangulate 2D tracks to form 3D tracks.
+        sfm_tracks, avg_track_reproj_errors, triangulation_exit_codes = self.run_triangulation(
```

Comment:
maybe we want to measure these two runtimes separately?

Reply:
Updated, thanks.